doc/nrf/libraries/others/trusted_storage.rst

```diff
@@ -0,0 +1,126 @@
+  Examples of assets that require this kind of storage are replay protection values for external storage and keys for use by components of the PSA Root of Trust.
+* Protected storage, designed to store all other critical data that do not need to be stored inside internal flash memory.
+
+Backends
+========
+
+The following backends are available in the trusted storage library:
+
+SECURE_STORAGE_BACKEND_AEAD
+   The ``SECURE_STORAGE_BACKEND_AEAD`` backend handles encryption, authentication or other means of validation of the stored data.
+   It is responsible for what is happening with the data before and after they are read from or written to non-volatile storage (flash memory).
+
+   The trusted storage library implements only ``SECURE_STORAGE_BACKEND_AEAD`` as a secure implementation backend, but it has support for adding other secure implementation backends to provide various levels of trust, depending on the device security features.
+
+SECURE_STORAGE_STORAGE_BACKEND_SETTINGS
+   This backend handles how the data are written to non-volatile storage.
+   It uses :ref:`Zephyr's settings subsystem <zephyr:settings_api>` for storing of data.
+
+   The trusted storage library implements only ``SECURE_STORAGE_STORAGE_BACKEND_SETTINGS`` as a storage backend, but it has support for adding other memory types for storage.
+
+
+Requirements
+************
+
+Use of hardware unique keys (HUKs) is required when using the trusted storage library.
+Derive and manage keys using the hardware unique key :ref:`library <lib_hw_unique_key>` and :ref:`sample <hw_unique_key_usage>`.
+
+Zephyr's settings subsystem has to be enabled for use?
+
+Configuration
+*************
+
+Set the Kconfig option :kconfig:option:`CONFIG_SECURE_STORAGE` to enable the trusted storage library.
+
+Once the library is enabled, two configuration options have to be set, :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND` and :kconfig:option:`CONFIG_SECURE_STORAGE_STORAGE_BACKEND`.
+
+Use the Kconfig option :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND` to define the backend that handles encryption and authentication.
+If this Kconfig option is set, the configuration defaults to the only currently available option :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD` to use an AEAD scheme for encryption and authentication of stored data.
+
+Use the Kconfig option :kconfig:option:`CONFIG_SECURE_STORAGE_STORAGE_BACKEND` to define the backend that handles how the data are written to and from the non-volatile storage.
+If this Kconfig option is set, the configuration defaults to the only currently available option :kconfig:option:`CONFIG_SECURE_STORAGE_STORAGE_BACKEND_SETTINGS` to use Zephyr's settings subsystem.
+
+The following options are used to configure the AEAD backend and its behavior:
+
+* :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_MAX_DATA_SIZE` - Defines the maximum data storage size for the AEAD backend (256 as default value).
```

Comment:
I see that some Kconfig options have BACKEND_AEAD and some AEAD_BACKEND in their name. Is this intentional? And if yes, why? :)

Reply:
also renamed it in #13833